pilot/pkg/bootstrap/server.go

```diff
@@ -110,13 +115,19 @@ type Server struct {
 	meshNetworks     *meshconfig.MeshNetworks
 	configController model.ConfigStoreCache
 
-	kubeClient            kubernetes.Interface
+	// Using Clientset because client is shared with other components - galley and few others expects Clientset.
+	// TODO: change everywhere to use Interface
+	kubeClientset *kubernetes.Clientset
+	kubeClient    kubernetes.Interface
+
 	startFuncs            []startFunc
 	multicluster          *clusterregistry.Multicluster
 	httpServer            *http.Server
 	grpcServer            *grpc.Server
 	secureHTTPServer      *http.Server
 	secureGRPCServer      *grpc.Server
+	secureHTTPServerDNS   *http.Server
+	secureGRPCServerDNS   *grpc.Server
```

Comment:
I have a doubt what if user enable this dns server with the original server simultaneously